mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,159 @@
+                greater than the number of layers in the input circuit.
+
+    """
+    if degree < 1:
+        raise ValueError(
+            "Multinomial degree must be greater than or equal to 1."
+        )
+    if fold_multiplier < 1:
+        raise ValueError("Fold multiplier must be greater than or equal to 1.")
+
+    scale_factor_vectors = _get_scale_factor_vectors(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    num_layers = len(scale_factor_vectors[0])
+
+    # Evaluate the monomial terms using the values in the scale factor vectors
+    # and insert in the sample matrix
+    # each row is specific to each scale factor vector
+    # each column is a term in the monomial basis
+    variable_exp = _full_monomial_basis_term_exponents(num_layers, degree)
+    sample_matrix = np.empty((len(variable_exp), len(variable_exp)))
+
+    for rows, i in enumerate(scale_factor_vectors):
+        for cols, j in enumerate(variable_exp):
+            evaluated_terms = []
+            for base, exp in zip(list(i), j):
+                # raise scale factor value by the exponent dict value
+                evaluated_terms.append(base**exp)
+            # multiply both elements in the list to create an evaluated
+            # monomial term
+            sample_matrix[rows, cols] = np.prod(evaluated_terms)
+
+    return sample_matrix
+
+
+def linear_combination_coefficients(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+) -> List[np.float64]:
+    r"""
+    Defines the sample matrix required for multivariate extrapolation as
+    defined in :cite:`Russo_2024_LRE`.
+
+    Args:
+        input_circuit: Circuit to be scaled.
+        degree: Degree of the multivariate polynomial.
+        fold_multiplier: Scaling gap required by unitary folding.
+        num_chunks: Number of desired approximately equal chunks. When the
+            number of chunks is the same as the layers in the input circuit,
+            the input circuit is unchanged.
+
+    Returns:
+        List of the evaluated monomial basis terms using the scale factor
+            vectors.
+    """
+    num_layers = len(
+        _get_scale_factor_vectors(
+            input_circuit, degree, fold_multiplier, num_chunks
+        )
+    )
+    input_sample_matrix = sample_matrix(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    try:
+        det = np.linalg.det(input_sample_matrix)
+    except RuntimeWarning:  # pragma: no cover
+        # taken from https://stackoverflow.com/a/19317237
+        warnings.warn(  # pragma: no cover
+            "To account for overflow error, required determinant of "
+            + "large sample matrix is calculated through "
+            + "`np.linalg.slogdet`."
+        )
+        sign, logdet = np.linalg.slogdet(  # pragma: no cover
+            input_sample_matrix
+        )
+        det = np.exp(logdet)  # pragma: no cover
```

Comment:
Need to keep the sign, right?
```suggestion
        det = sign * np.exp(logdet)  # pragma: no cover
```